src/main/java/com/openlattice/datastore/search/controllers/SearchController.java

```diff
@@ -141,8 +141,36 @@ public SearchResult getEntitySets(
             produces = { MediaType.APPLICATION_JSON_VALUE } )
     @Override
     public DataSearchResult searchEntitySetData( @RequestBody SearchConstraints searchConstraints ) {
-
-        DataSearchResult results = searchService.executeSearch( searchConstraints, Principals.getCurrentPrincipals() );
+        // check read on entity sets
+        final var allNormalEntitySetIds = Arrays.stream( searchConstraints.getEntitySetIds() )
```

Comment:
This logic doesn't look present in the PersistentSearchMessenger change in conductor-client. The permission flow here and there should be identical given a set of principals, so this might be something to add to EdmAuthorizationHelper too

Reply:
Thanks, my bad, will include entity set read checks there also.
I will add a different call, where it checks some permissions on entitysets, but I wouldn't include it in the same call, because that is supposed to check for propertytypes.

Reply:
sounds good!